apps/server/src/shared/repo/user/user.repo.ts

```diff
@@ -29,10 +29,7 @@ export class UserRepo extends BaseRepo<User> {
 			const { systems } = user.school;
 			return systems && systems.getItems().find((system) => system.id === systemId);
 		});
-		if (resultUser) {
-			return resultUser;
-		}
-		throw new NotFoundException('No user for this ldapId found');
+		return resultUser ?? Promise.reject();

```

Comment:
If this method fails instead of returning undefined when there is no user, please make that visibible in the function name. For example
`findByLdapIdOrFail`

Reply:
done